lnd-rpc/src/main/scala/org/bitcoins/lnd/rpc/config/LndInstance.scala

```diff
@@ -0,0 +1,87 @@
+package org.bitcoins.lnd.rpc.config
+
+import org.bitcoins.core.config._
+import org.bitcoins.rpc.config.BitcoindAuthCredentials._
+import org.bitcoins.rpc.config._
+import scodec.bits._
+
+import java.io.File
+import java.net._
+import java.nio.file._
+import scala.util.Properties
+
+case class LndInstance(
+    datadir: Path,
+    network: BitcoinNetwork,
+    listenBinding: URI,
+    restUri: URI,
+    rpcUri: URI,
+    bitcoindAuthCredentials: PasswordBased,
+    bitcoindRpcUri: URI,
+    zmqConfig: ZmqConfig,
+    debugLevel: LogLevel) {
+
+  lazy val certFile: File =
+    datadir.resolve("tls.cert").toFile
+
+  private var macaroonOpt: Option[String] = None
+
+  def macaroon: String = {
+    macaroonOpt match {
+      case Some(value) => value
+      case None =>
+        val path =
+          datadir
+            .resolve("data")
+            .resolve("chain")
+            .resolve("bitcoin")
+            .resolve(LndInstance.getNetworkDirName(network))
+            .resolve("admin.macaroon")
+
+        val bytes = Files.readAllBytes(path)
+        val hex = ByteVector(bytes).toHex
+
+        macaroonOpt = Some(hex)
+        hex
+    }
+  }
+}
+
+object LndInstance {
```

Comment:
Same thing here, it would be nice to have a common interface. 

I guess we don't have to figure it all out on this PR, at minimum though we should file an issue